erizo/src/erizo/media/ExternalOutput.cpp

```diff
@@ -470,32 +548,39 @@ void ExternalOutput::queueData(char* buffer, int length, packetType type) {
 }
 
 int ExternalOutput::sendFirPacket() {
-    if (fbSink_ != NULL) {
-      RtcpHeader thePLI;
-      thePLI.setPacketType(RTCP_PS_Feedback_PT);
-      thePLI.setBlockCount(1);
-      thePLI.setSSRC(55543);
-      thePLI.setSourceSSRC(videoSourceSsrc_);
-      thePLI.setLength(2);
-      char *buf = reinterpret_cast<char*>(&thePLI);
-      int len = (thePLI.getLength() + 1) * 4;
-      fbSink_->deliverFeedback(reinterpret_cast<char*>(buf), len);
-      return len;
-    }
-    return -1;
+  if (fbSink_ != NULL) {
+    RtcpHeader thePLI;
+    thePLI.setPacketType(RTCP_PS_Feedback_PT);
+    thePLI.setBlockCount(1);
+    thePLI.setSSRC(55543);
+    thePLI.setSourceSSRC(videoSourceSsrc_);
+    thePLI.setLength(2);
+    char *buf = reinterpret_cast<char*>(&thePLI);
+    int len = (thePLI.getLength() + 1) * 4;
+    fbSink_->deliverFeedback(reinterpret_cast<char*>(buf), len);
+    return len;
+  }
+  return -1;
 }
 
 void ExternalOutput::sendLoop() {
   while (recording_) {
     boost::unique_lock<boost::mutex> lock(mtx_);
     cond_.wait(lock);
-    while (audioQueue_.hasData()) {
-      boost::shared_ptr<dataPacket> audioP = audioQueue_.popPacket();
-      this->writeAudioData(audioP->data, audioP->length);
-    }
-    while (videoQueue_.hasData()) {
-      boost::shared_ptr<dataPacket> videoP = videoQueue_.popPacket();
-      this->writeVideoData(videoP->data, videoP->length);
+    if (hasAudio) {
+      while (audioQueue_.hasData() && audio_first_rtcp_time != 0 && video_first_rtcp_time != 0) {
```

Comment:
It works video + audio or video only. I haven't tested audio only. Thanks for your precious suggestions, in the next days I'll try to adjust these things.